template/src/pageComponents/mint/steps/StartMintStep.tsx

```diff
@@ -1,63 +1,93 @@
-import { useCallback } from 'react';
+import { useCallback, useEffect } from 'react';
 import clsx from 'clsx';
 import { TransactionExecutionError } from 'viem';
-import { useAccount, useContractWrite, useNetwork, usePrepareContractWrite } from 'wagmi';
+import {
+  useAccount,
+  useSimulateContract,
+  useWaitForTransactionReceipt,
+  useWriteContract,
+} from 'wagmi';
 import Button from '../../../components/Button/Button';
 import { EXPECTED_CHAIN } from '../../../constants';
 import { useCustom1155Contract } from '../../../hooks/contracts';
 import { MintSteps } from '../ContractDemo';
+import MintCompleteStep from './MintCompleteStep';
+import MintProcessingStep from './MintProcessingStep';
+import OutOfGasStep from './OutOfGasStep';
 
 type StartMintProps = {
   setMintStep: React.Dispatch<React.SetStateAction<MintSteps | null>>;
+  mintStep: MintSteps | null;
+  collectionName: string | null;
 };
 
-export default function StartMintStep({ setMintStep }: StartMintProps) {
-  const { chain } = useNetwork();
+export default function StartMintStep({ setMintStep, mintStep, collectionName }: StartMintProps) {
+  const { chain } = useAccount();
   const { address } = useAccount();
 
   const contract = useCustom1155Contract();
 
   const onCorrectNetwork = chain?.id === EXPECTED_CHAIN.id;
 
-  const { config } = usePrepareContractWrite({
+  const { data: mintData } = useSimulateContract({
     address: contract.status === 'ready' ? contract.address : undefined,
     abi: contract.abi,
     functionName: 'mint',
     args: address ? [address, BigInt(1), BigInt(1), address] : undefined,
-    enabled: onCorrectNetwork,
+    query: {
+      enabled: onCorrectNetwork,
+    },
   });
 
-  // A future enhancement would be to use the `isLoading` and `isSuccess`
-  // properties returned by `useContractWrite` to indicate transaction
-  // status in the UI.
-  const { write: performMint } = useContractWrite({
-    ...config,
-    onSuccess() {
-      setMintStep(MintSteps.MINT_COMPLETE_STEP);
+  const { writeContract: performMint, error: errorMint, data: dataMint } = useWriteContract();
+
+  const { status: transactionStatus } = useWaitForTransactionReceipt({
```

Comment:
Relying on status of `useWaitForTransactionReceipt` only means that there's an awkward delay for the user while`useSimulateContract` and `useWriteContract` are pending. What we do is combine all 3 statuses, and errors from all 3 hooks. It adds a lot of boilerplace code and repetition unfortunately and we didn't manage to create a re-usable hook yet to abstract that, due to generic types complexity, but it's a small tradeoff